Userland/Services/SystemServer/main.cpp

```diff
@@ -72,18 +72,19 @@ static ErrorOr<void> determine_system_mode()
             g_system_mode = "text";
     });
 
-    auto f = Core::DeprecatedFile::construct("/sys/kernel/constants/system_mode");
-    if (!f->open(Core::OpenMode::ReadOnly)) {
-        dbgln("Failed to read system_mode: {}", f->error_string());
+    auto f_or_error = Core::File::open("/sys/kernel/constants/system_mode"sv, Core::File::OpenMode::Read);
+    if (f_or_error.is_error()) {
+        dbgln("Failed to read system_mode: {}", f_or_error.error());
         // Continue and assume "text" mode.
         return {};
     }
-    const DeprecatedString system_mode = DeprecatedString::copy(f->read_all(), Chomp);
-    if (f->error()) {
-        dbgln("Failed to read system_mode: {}", f->error_string());
+    auto const system_mode_buf_or_error = f_or_error.value()->read_until_eof();
+    if (system_mode_buf_or_error.is_error()) {
+        dbgln("Failed to read system_mode: {}", system_mode_buf_or_error.error());
         // Continue and assume "text" mode.
         return {};
     }
+    const DeprecatedString system_mode = DeprecatedString::copy(system_mode_buf_or_error.value(), Chomp);
```

Comment:
Nit: east const

Reply:
Huh, weird. Usually I prefer east const, but in this particular case I didn't even notice that there was a choice. Thanks!